### PR TITLE
Allows Castlestation to be voted if revolutionaries won

### PR DESCRIPTION
**Pull request by boy2mantwicethefam**
Requires some secret repo configuration where the map folder is called "Castle Station", do not merge unless this is changed
Commented out Lamprey (condition is if the station has -20000 points or less) because Lamprey has a tendency to crash

:cl:
 * rscadd: Castlestation can now be voted if revolutionaries win.

### CODE REVIEW

**Comment by ghost** [comment]
holy based

**Comment by dunsel-hordon** [comment]
FUCKING YES. Actually gives antags that know about this a reason to absolutely fuck everything up

**Comment by jknpj** [comment]
isn't castle outdated

**Comment by Zth--** [comment]
does it even compile? You also need to add it to travis

**Comment by Pancakemines** [comment]
Is it possible to do this same kind of thing to make a map not in the vote at all? Such as if nukeops won make the map they just nuked skip that vote.

**Comment by chesse20** [comment]
lamprey crashing fits the flavor of the station dont remove please

**Comment by Kurfursten** [comment]
@boy2mantwicethefam If it was just a matter of secret repo we could merge it, this isn't a matter of secret repo - it's not something we keep on the secret repo. What is required is a compiled dmb file on pomf's computer.

Also, update Castle

**Comment by boy2mantwicethefam** [comment]
@Kurfursten are you sure it's not a matter of the secret repo? all I know is that the public repo and secret repo (that contains the server configs and everything) are supposed to overlap

**Comment by boy2mantwicethefam** [comment]
@chesse20 lamprey isn't being removed but the functionality to roll it is disabled for now

**Comment by Kurfursten** [comment]
We do not store any map files on the secret repo and none of the configs have to do with map availability. MoMMI automatically flags your PR if it touches secret repo overlapping files (there are very few, something like 5-6 total)

**Comment by boy2mantwicethefam** [comment]
shrug

File: code/__HELPERS/files.dm [hunk 1]
```diff
@@ -73,6 +73,22 @@
 				warning("Skipping map [potential] due to  it being too hot outside. Ideal temp is below 40F, found [temperature].")
 				binary = null
 				continue
+//Lamprey currently has a tendency to crash for no real reason, uncomment this when it stops being broken
+/*
+		if(potential == "Lamprey Station/") //Available if the station is wrecked enough
```
Comment (DamianX) [inline code/__HELPERS/files.dm:78]: delete all this

Comment (boy2mantwicethefam) [reply code/__HELPERS/files.dm:78]: This will be uncommented when Lamprey gets fixed, it's just convenience

**Comment by jknpj** [comment]
Castle won't compile. This PR can't be merged until it is fixed.

**Comment by Zth--** [comment]
And it still needs to be added to travis

**Comment by boy2mantwicethefam** [comment]
Castle compiles as of #27140 and is also added to Travis as of #27140

**Comment by boy2mantwicethefam** [comment]
Castle was tested and it had no visible issue, however it had a bunch of outdated things:
- Doesn't have the big maintenance that displaced the pod bay
- Doesn't have a shutter in the captain's office maintenance
- Doesn't have a labor console in the head of personnel's office

**Comment by DeityLink** [comment]
wouldn't the easiest way to keep Castle up to date be to just always re-use the current version of box and copy-paste the removed sec?

**Comment by boy2mantwicethefam** [comment]
Is it even possible to copypaste between maps?

**Comment by Kurfursten** [comment]
yes

**Comment by DeityLink** [comment]
@boy2mantwicethefam Do you really think I gave every map the same Centcom tile per tile?

**Comment by boy2mantwicethefam** [comment]
yes

**Comment by juanototo2** [comment]
Boy2man please finish this PR for the love of God

**Comment by Kurfursten** [comment]
Barry said it's tested. If it doesn't work, someone execute him.

**Comment by dunsel-hordon** [comment]
i mean how long is gonna take to actually get -20000 in a round

**Comment by ghost** [comment]
lamprey is commented out of this PR so that part isn't a problem

**Comment by mrflarg** [comment]
blessed Albanian

**Pull request merged**